bundles/org.openhab.binding.magentatv/src/main/java/org/openhab/binding/magentatv/internal/handler/MagentaTVHandler.java

```diff
@@ -0,0 +1,688 @@
+                    control.sendKey("POWER");
+                    break;
+                case CHANNEL_PLAYER:
+                    logger.debug("{}: Player command: {}", thingId, command);
+                    if (command instanceof OnOffType) {
+                        control.sendKey("POWER");
+                    } else if (command instanceof PlayPauseType) {
+                        if (command == PlayPauseType.PLAY) {
+                            control.sendKey("PLAY");
+                        } else if (command == PlayPauseType.PAUSE) {
+                            control.sendKey("PAUSE");
+                        }
+                    } else if (command instanceof NextPreviousType) {
+                        if (command == NextPreviousType.NEXT) {
+                            control.sendKey("NEXTCH");
+                        } else if (command == NextPreviousType.PREVIOUS) {
+                            control.sendKey("PREVCH");
+                        }
+                    } else if (command instanceof RewindFastforwardType) {
+                        if (command == RewindFastforwardType.FASTFORWARD) {
+                            control.sendKey("FORWARD");
+                        } else if (command == RewindFastforwardType.REWIND) {
+                            control.sendKey("REWIND");
+                        }
+                    } else {
+                        logger.debug("{}: Unknown media command: {}", thingId, command);
+                    }
+                    break;
+                case CHANNEL_CHANNEL:
+                    String chan = command.toString();
+                    control.selectChannel(chan);
+                    break;
+                case CHANNEL_MUTE:
+                    if (command == OnOffType.ON) {
+                        control.sendKey("MUTE");
+                    } else {
+                        control.sendKey("VOLUP");
+                    }
+                    break;
+                case CHANNEL_KEY:
+                    if (command.toString().equalsIgnoreCase("PAIR")) { // special key to re-pair receiver (already done
+                                                                       // above)
+                        logger.debug("{}: PAIRing key received, reconnect receiver {}", thingId, deviceName());
+                    } else {
+                        control.sendKey(command.toString());
+                        mapKeyToMediateState(command.toString());
+                    }
+                    break;
+                default:
+                    logger.debug("{}: Command {} for unknown channel {}", thingId, command, channelUID.getAsString());
+            }
+        } catch (MagentaTVException e) {
+            String errorMessage = MessageFormat.format("Channel operation failed (command={0}, value={1}): {2}",
+                    command, channelUID.getId(), e.toString());
```

Comment:
The exception toString also contains the whole stacktrace, so you probalby what to do `e.getMessage()`

Reply:
yep, changed